students/EricAdams/session07/html_render.py

```diff
@@ -35,12 +40,45 @@ def write_to_file(self, file_obj, stuff_to_print):
 
 class Body(Element):
     tag = 'body'
+    indent = '    '
 
 
 class P(Element):
     tag = 'p'
+    indent = '    '
 
 
 class Html(Element):
     tag = 'html'
+    indent = ''
+
+
+class Head(Element):
+    tag = 'head'
+    indent = '    '
+
+
+class OnelineTag(Element):
+    # override Element.render to write all on one line
+    def render(self, file_obj, indent=''):
+        # what recursion level are we going in
+        Element.depth += 1
+        html_tag = (self.indent * (Element.depth - 1) + '<' + self.tag + '>')
+        file_obj.write(html_tag)
+        text = ""
+        for each in self.content:
+            try:
+                text += each
```

Comment:
this is where things can get out of order -- better to render either the text or the sub-element to the file in the same loop.

Reply:
though it is working, yes?

but did you test arbitrary mixing of sub-elements and text?